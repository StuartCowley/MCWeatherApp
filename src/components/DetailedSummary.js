import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";

import "../styles/DetailedSummary.css";

const DetailedSummary = ({ forecast }) => {
  const { date, description, humidity, icon, temperature, wind } = forecast;

  // Create date in format `shortWeekDay numericDay shortMonth` for local language (hard coded to GB for now)
  const locale = "en-GB";
  const options = {
    month: "short",
    day: "numeric",
  };
  // As toLocaleDateString() includes a comma after the day by default,
  // we have to get the day first then append the numeric day and month
  let formattedDate = new Date(date).toLocaleDateString(locale, {
    weekday: "short",
  });
  formattedDate += ` ${new Date(date).toLocaleDateString(locale, options)}`;

  return (
    <div className="detailed-summary" data-testid="detailed-summary">
      <div className="detailed-summary__date">{formattedDate}</div>
      <WeatherIcon
        data-testid="forecast-icon"
        className="detailed-summary__icon"
        name="owm"
        iconId={icon}
        flip="horizontal"
        rotate="90"
      />
      <div className="detailed-summary__description">{description}</div>
      <div className="detailed-summary__humidity">{humidity}</div>
      <div className="detailed-summary__min-temp">{temperature.min}</div>
      <div className="detailed-summary__wind-speed">{wind.speed}</div>
      <div className="detailed-summary__wind-direction">{wind.direction}</div>
    </div>
  );
};

DetailedSummary.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    description: PropTypes.string,
    humidity: PropTypes.number,
    icon: PropTypes.string,
    temperature: PropTypes.shape({
      min: PropTypes.number,
    }),
    wind: PropTypes.shape({
      direction: PropTypes.string,
      speed: PropTypes.number,
    }),
  }).isRequired,
};

export default DetailedSummary;
