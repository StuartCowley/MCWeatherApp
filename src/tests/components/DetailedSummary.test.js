import React from "react";
import { render } from "@testing-library/react";
import DetailedSummary from "../../components/DetailedSummary";

describe("DetailedSummary", () => {
  const validProps = {
    date: 1525046400000,
    description: "Cloudy",
    icon: "300",
    temperature: {
      min: 10,
      max: 15,
    },
    wind: { speed: 13, direction: "s" },
    humidity: 22,
  };

  it("renders correctly", () => {
    const { asFragment } = render(<DetailedSummary forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <DetailedSummary forecast={validProps} />
    );
    expect(getByText("Cloudy")).toHaveClass("detailed-summary__description");
    expect(getByTestId("forecast-icon")).toHaveClass("detailed-summary__icon");
    expect(getByText(22)).toHaveClass("detailed-summary__humidity");
    expect(getByText("Mon 30 Apr")).toHaveClass("detailed-summary__date");
    expect(getByText(`10\u2103`)).toHaveClass("detailed-summary__min-temp");
    expect(getByText(13)).toHaveClass("detailed-summary__wind-speed");
    expect(getByText("s")).toHaveClass("detailed-summary__wind-direction");
  });
});
