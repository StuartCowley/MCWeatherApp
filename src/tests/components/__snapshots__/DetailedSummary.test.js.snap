// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="detailed-summary"
    data-testid="detailed-summary"
  >
    <div
      class="detailed-summary__date"
    >
      Mon 30 Apr
    </div>
    <i
      class="wi wi-day-sprinkle wi-flip-horizontal wi-rotate-90 detailed-summary__icon"
      data-testid="forecast-icon"
      name="owm"
    />
    <div
      class="detailed-summary__max-temp"
    >
      15â„ƒ
    </div>
    <div
      class="detailed-summary__description"
    >
      Cloudy
    </div>
    <div
      class="detailed-summary__humidity"
    >
      22
    </div>
    <div
      class="detailed-summary__wind-speed"
    >
      13
    </div>
    <div
      class="detailed-summary__wind-direction"
    >
      s
    </div>
  </div>
</DocumentFragment>
`;
